### DISCUSSION TITLE
Feature Request: External Scripts Integration and Publishing Tools

### DISCUSSION THREAD

**Discussion by snden** [vkbo/novelWriter#2083] [Ideas]
Dear Veronica,

I'm greatly appreciates your software for its excellent project organization capabilities. I've created a video showcasing some feature suggestions that I believe could enhance NovelWriter's functionality.

The main ideas include:
- Batch text processing using external scripts (e.g., Python scripts)
- Character count features specific for Czech publishing standards (1800 characters per page)
- Right-click formatting options for italic/bold text and quotation marks
- Better handling of image placeholders in the text and export

I've developed a simple Python application that demonstrates these concepts, which I explain in the video:
https://youtu.be/VKvouLYhl5Q
The YouTube video is not publicly visible but can be accessed through the provided link.

I apologize that the video commentary is in Czech, as I'm not confident enough to present fluently in English. However, I've added subtitles that can be automatically translated by YouTube into English or other languages. I hope this makes the content accessible despite the language barrier.

I believe adding support for external scripts/plugins could make NovelWriter even more versatile while maintaining its core simplicity. These suggestions come from my experience as an amateur writer and  programmer. If you decide to implement external script support in the future, I'd be happy to share my sample scripts.

Thank you for creating this excellent writing tool. Your work has significantly improved my writing process, and I hope these suggestions might be useful for future development.

Best regards,
Pavel

**Comment by vkbo** [comment]
I haven't looked at your video yet, I dislike videos as a medium and YouTube in particular, but I'll try to get to it. That said, for any feature request to be considered, it must be added here as a proper text ticket,

Let me just briefly respond to your points.

> * Batch text processing using external scripts (e.g., Python scripts)

While I am a programmer, and I would expect novelWriter appeals to more technically oriented users being plain text and all, I do not plan to add programmer focused tooling beyond a few simple features. One that already exists is the JSON export files, which is intended for post-processing with code. The other is that I do intend to add command line support for the manuscript build tool. That's why I added the build settings feature based on JSON config files. The idea is that you could write a Python or Bash script to quickly generate manuscripts from one or more projects.

I do plan to add this, but it's still a bit down the line as GUI features have priority.

> * Character count features specific for Czech publishing standards (1800 characters per page)

Could you explain what that would entail? The new Manuscript stats feature shows pretty much every statistic I could think of. Page count is a rather arbitrary unit, so it's hard to generalise.

> * Right-click formatting options for italic/bold text and quotation marks

That is an unusual request? It has support for the standard shortcuts, and a toolbar for those who prefer point-and-click. Context menus are generally annoying to deal with both from a programming and memory management point of view,

> * Better handling of image placeholders in the text and export

There is no support for images, and there are no plans to add it. When I add custom notes, soon, you can add specific notes for where images should be placed, and you should be able to filter on those.

> Thank you for creating this excellent writing tool. Your work has significantly improved my writing process, and I hope these suggestions might be useful for future development.

I'm glad you find it useful, even without these features!

**Reply by vkbo** [reply]
Ok, I watched the first part of the video. I now see what you mean by batch processing. I do plan to add text processing features, yes. I want to add a panel below the editor where you can select various tools to check and clean up text.
